assignment to entry in nil map

The following panic occurs when creating a PyTorchJob without annotations:

```text
E0119 09:13:28.543552       1 runtime.go:73] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
goroutine 154 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x13f4fc0, 0x1797d60)
    /go/pkg/mod/k8s.io/apimachinery@v0.15.10-beta.0/pkg/util/runtime/runtime.go:69 +0x7b
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
    /go/pkg/mod/k8s.io/apimachinery@v0.15.10-beta.0/pkg/util/runtime/runtime.go:51 +0x82
panic(0x13f4fc0, 0x1797d60)
    /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch.(*PyTorchController).deletePodsAndServices(0xc000549e00, 0xc001062c00, 0xc0011e3e60, 0x2, 0x2
    /go/src/github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch/job.go:192 +0x322
github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch.(*PyTorchController).reconcilePyTorchJobs(0xc000549e00, 0xc001062c00, 0xc001062c00, 0xc000264
    /go/src/github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch/controller.go:382 +0x5be
github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch.(*PyTorchController).syncPyTorchJob(0xc000549e00, 0xc000264d60, 0x1d, 0x0, 0x0, 0x0)
    /go/src/github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch/controller.go:343 +0x4a2
github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch.(*PyTorchController).processNextWorkItem(0xc000549e00, 0x224bf00)
    /go/src/github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch/controller.go:281 +0x56f
github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch.(*PyTorchController).runWorker(0xc000549e00)
    /go/src/github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch/controller.go:226 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000754780)
    /go/pkg/mod/k8s.io/apimachinery@v0.15.10-beta.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000754780, 0x3b9aca00, 0x0, 0x1, 0xc0000ac120)
    /go/pkg/mod/k8s.io/apimachinery@v0.15.10-beta.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000754780, 0x3b9aca00, 0xc0000ac120)
    /go/pkg/mod/k8s.io/apimachinery@v0.15.10-beta.0/pkg/util/wait/wait.go:88 +0x4d
created by github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch.(*PyTorchController).Run
    /go/src/github.com/kubeflow/pytorch-operator/pkg/controller.v1/pytorch/controller.go:212 +0x2c4
{"filename":"record/event.go:258","level":"info","msg":"Event(v1.ObjectReference{Kind:\"PyTorchJob\", Namespace:\"default\", Name:\"cron-allow-1768813680\",
panic: assignment to entry in nil map [recovered]
    panic: assignment to entry in nil map
```

--- a/pkg/controller.v1/pytorch/job.go
+++ b/pkg/controller.v1/pytorch/job.go
@@ -189,6 +189,9 @@ func (pc *PyTorchController) deletePodsAndServices(job *pyv1.PyTorchJob, pods []
 	}
 
 	jobToUpdate := job.DeepCopy()
+	if jobToUpdate.Annotations == nil {
+		jobToUpdate.Annotations = map[string]string{}
+	}
 	jobToUpdate.Annotations[PytorchCleanPodStatusLabel] = PytorchCleanStatusDone
 	if !reflect.DeepEqual(job, jobToUpdate) {
 		_, err := pc.jobClientSet.KubeflowV1().PyTorchJobs(jobToUpdate.Namespace).Update(jobToUpdate)

